### ISSUE TITLE
Version 0.9.88 causes installation failures in Docker

### ISSUE THREAD

**Issue by JamesChevalier** [oxidize-rb/rb-sys#330]
Here's the smallest reproduction I could come up with.

Dockerfile:

```
FROM ruby:3.3
WORKDIR /app
RUN apt-get update -qq && \
    apt-get upgrade -y && \
    apt-get install -y libclang-dev
COPY Gemfile.issue /app/Gemfile
RUN bundle install
```

Gemfile:

```
source "https://rubygems.org"
gem "rb_sys", "0.9.88"
gem "oxi-test", git: "https://github.com/oxidize-rb/oxi-test", branch: :main
```

The `docker build` attempt fails with a "make: cargo: No such file or directory" error.

<details>
<summary>Full error output</summary>

```
 > [5/5] RUN bundle install:                                                                                                                                    
0.217 Fetching https://github.com/oxidize-rb/oxi-test
1.815 Fetching gem metadata from https://rubygems.org/.
1.848 Resolving dependencies...
1.860 Fetching rb_sys 0.9.88
1.982 Installing rb_sys 0.9.88
2.045 Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
2.045 
2.045     current directory: /usr/local/bundle/bundler/gems/oxi-test-1528361e233b/ext
2.045 /usr/local/bin/ruby extconf.rb
2.045 checking for gcc... yes
2.045 checking for g++... yes
2.045 checking for gcc-ar... yes
2.045 
2.045 current directory: /usr/local/bundle/bundler/gems/oxi-test-1528361e233b/ext
2.045 make DESTDIR\= sitearchdir\=./.gem.20240221-7-lhs6z5
2.045 sitelibdir\=./.gem.20240221-7-lhs6z5 clean
2.045 
2.045 current directory: /usr/local/bundle/bundler/gems/oxi-test-1528361e233b/ext
2.045 make DESTDIR\= sitearchdir\=./.gem.20240221-7-lhs6z5
2.045 sitelibdir\=./.gem.20240221-7-lhs6z5
2.045 generating target/release/liboxi_test.so (release)
2.045 cargo rustc  --manifest-path ./Cargo.toml --target-dir target --lib --profile
2.045 release -- -C linker=gcc -L native=/usr/local/lib -C link-arg=-lm -l pthread
2.045 make: cargo: No such file or directory
2.045 make: *** [Makefile:566: target/release/liboxi_test.so] Error 127
2.045 
2.045 make failed, exit code 2
2.045 
2.045 Gem files will remain installed in
2.045 /usr/local/bundle/bundler/gems/oxi-test-1528361e233b for inspection.
2.045 Results logged to
2.045 /usr/local/bundle/bundler/gems/extensions/aarch64-linux/3.3.0/oxi-test-1528361e233b/gem_make.out
2.045 
2.045   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:125:in `run'
2.045   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:51:in `block in make'
2.045   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:43:in `each'
2.045   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:43:in `make'
2.045   /usr/local/lib/ruby/3.3.0/rubygems/ext/ext_conf_builder.rb:42:in `build'
2.045   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:193:in `build_extension'
2.045 /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:227:in `block in
2.045 build_extensions'
2.045   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in `each'
2.045   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in `build_extensions'
2.045   /usr/local/lib/ruby/3.3.0/rubygems/installer.rb:852:in `build_extensions'
2.045 /usr/local/lib/ruby/3.3.0/bundler/rubygems_gem_installer.rb:76:in
2.045 `build_extensions'
2.045 /usr/local/lib/ruby/3.3.0/bundler/source/path/installer.rb:28:in
2.045 `post_install'
2.045   /usr/local/lib/ruby/3.3.0/bundler/source/path.rb:245:in `generate_bin'
2.045   /usr/local/lib/ruby/3.3.0/bundler/source/git.rb:202:in `install'
2.045   /usr/local/lib/ruby/3.3.0/bundler/installer/gem_installer.rb:54:in `install'
2.045 /usr/local/lib/ruby/3.3.0/bundler/installer/gem_installer.rb:16:in
2.045 `install_from_spec'
2.045 /usr/local/lib/ruby/3.3.0/bundler/installer/parallel_installer.rb:132:in
2.045 `do_install'
2.045 /usr/local/lib/ruby/3.3.0/bundler/installer/parallel_installer.rb:123:in
2.045 `block in worker_pool'
2.045   /usr/local/lib/ruby/3.3.0/bundler/worker.rb:62:in `apply_func'
2.045   /usr/local/lib/ruby/3.3.0/bundler/worker.rb:57:in `block in process_queue'
2.045   <internal:kernel>:187:in `loop'
2.045   /usr/local/lib/ruby/3.3.0/bundler/worker.rb:54:in `process_queue'
2.045 /usr/local/lib/ruby/3.3.0/bundler/worker.rb:90:in `block (2 levels) in
2.045 create_threads'
2.045 
2.045 An error occurred while installing oxi-test (0.1.0), and Bundler cannot
2.045 continue.
2.045 
2.045 In Gemfile:
2.045   oxi-test
------
Dockerfile:7
--------------------
   5 |         apt-get install -y libclang-dev
   6 |     COPY Gemfile /app/
   7 | >>> RUN bundle install
   8 |     CMD echo 'Hello, world!'
   9 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c bundle install" did not complete successfully: exit code: 5
```
</details>

Changing the Gemfile contents from `gem "rb_sys", "0.9.88"` to `gem "rb_sys", "0.9.87"` results in a successful build.

I originally reported this in https://github.com/HarlemSquirrel/tzf-rb/issues/13 which includes some more examples of successful/unsuccessful builds.

**Comment by ianks** [comment]
Fixed in `v0.9.89`

**Comment by JamesChevalier** [comment]
Thanks for working on this! Unfortunately, I'm now getting a different error. Here are all the details...

Dockerfile:

```
FROM ruby:3.3
WORKDIR /app
RUN apt-get update -qq && \
    apt-get upgrade -y && \
    apt-get install -y libclang-dev
COPY Gemfile /app/
RUN bundle install
```

Gemfile:

```
source "https://rubygems.org"
ruby "3.3"
gem "rb_sys", github: "oxidize-rb/rb-sys"
gem "tzf"
```

<details>
<summary>Full error</summary>

```
 > [5/5] RUN bundle install:                                                                                                                                    
0.214 Fetching https://github.com/oxidize-rb/rb-sys.git                                                                                                         
2.550 Fetching gem metadata from https://rubygems.org/...                                                                                                       
2.708 Resolving dependencies...                                                                                                                                 
2.743 Fetching tzf 0.4.5
2.866 Installing tzf 0.4.5 with native extensions
2.928 Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
2.928 
2.928     current directory: /usr/local/bundle/gems/tzf-0.4.5/ext/tzf
2.928 /usr/local/bin/ruby extconf.rb
2.928 *** extconf.rb failed ***
2.928 Could not create Makefile due to some reason, probably lack of necessary
2.928 libraries and/or headers.  Check the mkmf.log file for more details.  You may
2.928 need configuration options.
2.928 
2.928 Provided configuration options:
2.928   --with-opt-dir
2.928   --without-opt-dir
2.928   --with-opt-include=${opt-dir}/include
2.928   --without-opt-include
2.928   --with-opt-lib=${opt-dir}/lib
2.928   --without-opt-lib
2.928   --with-make-prog
2.928   --without-make-prog
2.928   --srcdir=.
2.928   --curdir
2.928   --ruby=/usr/local/bin/$(RUBY_BASE_NAME)
2.928 <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in
2.928 `require': cannot load such file -- rb_sys/mkmf (LoadError)
2.928 from
2.928 <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in
2.928 `require'
2.928   from extconf.rb:4:in `<main>'
2.928 
2.928 extconf failed, exit code 1
2.928 
2.928 Gem files will remain installed in /usr/local/bundle/gems/tzf-0.4.5 for
2.928 inspection.
2.928 Results logged to
2.928 /usr/local/bundle/extensions/aarch64-linux/3.3.0/tzf-0.4.5/gem_make.out
2.928 
2.928   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:125:in `run'
2.928   /usr/local/lib/ruby/3.3.0/rubygems/ext/ext_conf_builder.rb:28:in `build'
2.928   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:193:in `build_extension'
2.928 /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:227:in `block in
2.928 build_extensions'
2.928   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in `each'
2.928   /usr/local/lib/ruby/3.3.0/rubygems/ext/builder.rb:224:in `build_extensions'
2.928   /usr/local/lib/ruby/3.3.0/rubygems/installer.rb:852:in `build_extensions'
2.928 /usr/local/lib/ruby/3.3.0/bundler/rubygems_gem_installer.rb:76:in
2.928 `build_extensions'
2.928   /usr/local/lib/ruby/3.3.0/bundler/rubygems_gem_installer.rb:28:in `install'
2.928   /usr/local/lib/ruby/3.3.0/bundler/source/rubygems.rb:205:in `install'
2.928   /usr/local/lib/ruby/3.3.0/bundler/installer/gem_installer.rb:54:in `install'
2.928 /usr/local/lib/ruby/3.3.0/bundler/installer/gem_installer.rb:16:in
2.928 `install_from_spec'
2.928 /usr/local/lib/ruby/3.3.0/bundler/installer/parallel_installer.rb:132:in
2.928 `do_install'
2.928 /usr/local/lib/ruby/3.3.0/bundler/installer/parallel_installer.rb:123:in
2.928 `block in worker_pool'
2.928   /usr/local/lib/ruby/3.3.0/bundler/worker.rb:62:in `apply_func'
2.928   /usr/local/lib/ruby/3.3.0/bundler/worker.rb:57:in `block in process_queue'
2.928   <internal:kernel>:187:in `loop'
2.928   /usr/local/lib/ruby/3.3.0/bundler/worker.rb:54:in `process_queue'
2.928 /usr/local/lib/ruby/3.3.0/bundler/worker.rb:90:in `block (2 levels) in
2.928 create_threads'
2.928 
2.928 An error occurred while installing tzf (0.4.5), and Bundler cannot continue.
2.928 
2.928 In Gemfile:
2.928   tzf
------
Dockerfile:7
--------------------
   5 |         apt-get install -y libclang-dev
   6 |     COPY Gemfile /app/
   7 | >>> RUN bundle install
   8 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c bundle install" did not complete successfully: exit code: 5
```
</details>
